librad/src/net/gossip.rs

```diff
@@ -265,9 +271,10 @@ impl<S: Clone, R, W> Clone for Protocol<S, R, W> {
     }
 }
 
-impl<S, R, W> Protocol<S, R, W>
+impl<S, A, R, W> Protocol<S, A, R, W>
 where
-    S: LocalStorage + 'static,
+    S: LocalStorage<Update = A> + 'static,
+    for<'de> A: Serialize + Deserialize<'de> + Clone + Debug + Send + Sync + 'static,
```

Comment:
Yikes, what's the `Debug` for?

Reply:
Because of `trace!`s -- that's perhaps an indication that these things should be treated first-class (I think I lamented about this before). Although intended for slightly different kinds of applications, I was wondering if we should settle on this: https://docs.rs/tracing/0.1.13/tracing/

Reply:
Ah yes, we did have a chat about this. I don't have strong opinions about this. Maybe we need a radicle-decision? :stuck_out_tongue:

Reply:
Not sure this applies to anyone but us (+ upstream as a consumer). Created #81